### PR TITLE
ci: support rxjs eslint plugin

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the new behavior?

https://github.com/cartant/eslint-plugin-rxjs

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/cdk/tokens/active-element.ts [hunk 1]
```diff
@@ -35,13 +35,15 @@ export const TUI_ACTIVE_ELEMENT = new InjectionToken<Observable<EventTarget | nu
...
-                    repeatWhen(() => mouseup$),
+                    repeatWhen(notifications =>
+                        notifications.pipe(switchMap(() => mouseup$)),
+                    ),
                     withLatestFrom(removedElement$),
                     filter(([event, removedElement]) =>
                         isValidFocusout(getActualTarget(event), removedElement),
                     ),
                     map(([{relatedTarget}]) => relatedTarget),
+                    takeUntil(mousedown$),
```
Comment (nsbarsukov) [inline projects/cdk/tokens/active-element.ts:46]: I am not sure but this can be mistake

Comment (splincode) [reply projects/cdk/tokens/active-element.ts:46]: No, this does not affect the order, also e2e test shows that there was misbehavior before

**Comment by splincode** [comment]
The screenshot fell correctly, this is the expected behavior!

**Review by nsbarsukov — APPROVED** [review]

File: projects/addon-mobile/directives/ripple/ripple.directive.ts [hunk 2]
```diff
@@ -118,6 +117,7 @@ export class TuiRippleDirective {
                                   ),
                         ),
                         mapTo(ripple),
+                        takeUntil(touchMove$),
```
[the remark this answers is not in the document]
Comment (vladimirpotekhin) [reply projects/addon-mobile/directives/ripple/ripple.directive.ts:120]: I think rule "takeUntil at the end of pipe" actual only for case takeUntil(**this.destroy$**)

**Review by vladimirpotekhin — CHANGES_REQUESTED** [review]

File: projects/addon-editor/components/editor-new/editor-new.providers.ts [hunk 3]
```diff
@@ -38,7 +38,9 @@ export const TUI_EDITOR_NEW_PROVIDERS = [
 export function extensionsFactory(
     extensions: Promise<Extension>[],
 ): Observable<ReadonlyArray<Extension | Mark | Node>> {
-    const extensions$ = new ReplaySubject<ReadonlyArray<Extension | Mark | Node>>();
+    const extensions$ = new ReplaySubject<ReadonlyArray<Extension | Mark | Node>>(
+        Number.POSITIVE_INFINITY,
```
Comment (waterplea) [inline projects/addon-editor/components/editor-new/editor-new.providers.ts:42]: I believe it should be 1 here

File: projects/addon-doc/src/components/example/example.component.ts [hunk 4]
```diff
@@ -57,7 +57,7 @@ export class TuiDocExampleComponent {
 
     readonly processor$: Observable<Record<string, string>> = this.rawLoader$$.pipe(
         switchMap(rawLoadRecord),
-        map(this.processContent),
+        map((value: Record<string, string>) => this.processContent(value)),
```
Comment (waterplea) [inline projects/addon-doc/src/components/example/example.component.ts:60]: Unnecessary type

File: projects/addon-mobile/directives/ripple/ripple.directive.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-mobile/directives/ripple/ripple.directive.ts]: fixed

[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-mobile/directives/ripple/ripple.directive.ts]: fixed

File: projects/cdk/services/pan.service.ts [hunk 5]
```diff
@@ -37,13 +37,13 @@ export class TuiPanService extends Observable<readonly [number, number]> {
...
                         return [deltaX, deltaY] as [number, number];
                     }),
+                    repeat(),
                     takeUntil(
                         merge(
                             typedFromEvent(documentRef, 'touchend'),
                             typedFromEvent(documentRef, 'mouseup'),
                         ),
                     ),
-                    repeat(),
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/cdk/services/pan.service.ts]: fixed

[the remark this answers is not in the document]
Comment (splincode) [reply projects/cdk/services/pan.service.ts]: fixed

File: projects/cdk/tokens/active-element.ts (hunk 1 above)
Comment (splincode) [reply projects/cdk/tokens/active-element.ts]: reverted

File: projects/addon-editor/tokens/tiptap-editor.ts [hunk 6]
```diff
@@ -8,7 +8,7 @@ export const TIPTAP_EDITOR = new InjectionToken<Observable<Editor>>(
 
 export const LAZY_TIPTAP_EDITOR = new InjectionToken('Lazy loaded Editor', {
     factory: () => {
-        const editor$ = new ReplaySubject<typeof Editor>();
+        const editor$ = new ReplaySubject<typeof Editor>(Number.POSITIVE_INFINITY);
```
Comment (waterplea) [inline projects/addon-editor/tokens/tiptap-editor.ts:11]: 1?

File: projects/cdk/services/static-request.service.ts [hunk 7]
```diff
@@ -38,7 +38,7 @@ export class TuiStaticRequestService {
                 xhr.abort();
             };
         });
-        const piped = observable.pipe(shareReplay(1));
+        const piped = observable.pipe(shareReplay({bufferSize: 1, refCount: true}));
```
Comment (waterplea) [inline projects/cdk/services/static-request.service.ts:41]: I think this is intentional, it replays last value even after it was complete and then resubscribed, `refCount` will make it drop last value if nobody is subscribed.

File: projects/cdk/services/zoom.service.ts [hunk 8]
```diff
@@ -56,6 +55,7 @@ export class TuiZoomService extends Observable<TuiZoom> {
 
                                 return {clientX, clientY, delta, event};
                             }),
+                            takeUntil(typedFromEvent(nativeElement, 'touchend')),
```
Comment (waterplea) [inline projects/cdk/services/zoom.service.ts:58]: Let's move `preventDefault` from above to a separate utility operator we have in cdk?

File: projects/cdk/tokens/active-element.ts [hunk 9]
```diff
@@ -35,8 +35,11 @@ export const TUI_ACTIVE_ELEMENT = new InjectionToken<Observable<EventTarget | nu
 
             return merge(
                 focusout$.pipe(
+                    // eslint-disable-next-line rxjs/no-unsafe-takeuntil
                     takeUntil(mousedown$),
-                    repeatWhen(() => mouseup$),
+                    repeatWhen(notifications =>
```
Comment (waterplea) [inline projects/cdk/tokens/active-element.ts:40]: Can you explain the difference?

File: projects/addon-doc/src/components/example/example.component.ts (hunk 4 above)
Comment (splincode) [reply projects/addon-doc/src/components/example/example.component.ts:60]: but then the rule breaks

image

File: projects/cdk/tokens/active-element.ts (hunk 9 above)
Comment (splincode) [reply projects/cdk/tokens/active-element.ts:40]: This is recommended rule 

image

it also revealed an incorrect e2e test

File: projects/addon-editor/components/editor-new/editor-new.providers.ts (hunk 3 above)
Comment (splincode) [reply projects/addon-editor/components/editor-new/editor-new.providers.ts]: fixed

File: projects/addon-editor/tokens/tiptap-editor.ts (hunk 6 above)
Comment (splincode) [reply projects/addon-editor/tokens/tiptap-editor.ts]: fixed

File: projects/cdk/services/static-request.service.ts (hunk 7 above)
Comment (splincode) [reply projects/cdk/services/static-request.service.ts]: fixed

File: projects/cdk/services/zoom.service.ts (hunk 8 above)
Comment (splincode) [reply projects/cdk/services/zoom.service.ts:57]: Unfortunately, the type of the `preventDefault ` operator does not match the value returned from the stream

Comment (waterplea) [reply projects/cdk/services/zoom.service.ts:57]: You just need to move it up before `scan` :)

File: projects/addon-doc/src/components/example/example.component.ts (hunk 4 above)
Comment (waterplea) [reply projects/addon-doc/src/components/example/example.component.ts:60]: I mean the type is unnecessary, I'm ok with following the rule :)

**Comment by waterplea** [comment]
I think in most cases we can use `refCount: true`, just gotta be sure it's right :)

File: projects/cdk/tokens/active-element.ts (hunk 9 above)
Comment (waterplea) [reply projects/cdk/tokens/active-element.ts:40]: I just don't know how this notifier works :)

File: projects/addon-doc/src/components/example/example.component.ts (hunk 4 above)
Comment (splincode) [reply projects/addon-doc/src/components/example/example.component.ts:60]: fixed

**Comment by splincode** [comment]
> I think in most cases we can use refCount: true, just gotta be sure it's right :)

Added where needed

**Review by waterplea — APPROVED** [review]

**Review by nsbarsukov — APPROVED** [review]

**Comment by splincode** [comment]
@vladimirpotekhin hello, please, review me)

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**